src/main/src/main/java/org/geoserver/security/GeoServerSecurityManager.java

```diff
@@ -1930,6 +1930,9 @@ void writeMasterPasswordInfo(Resource file, String message, char[] masterPasswor
             w.write("Test the master password by logging in as user \"root\"");
             w.newLine();
             w.newLine();
+            w.write("Note: Login with the master password must be enabled in security/masterpw/default/config.xml");
```

Comment:
There is a way to enable this using the user interface, hand editing config.xml is not necessary or recommended.

You could link to the appropriate UI setting...

Reply:
The config.xml method is mentioned on https://docs.geoserver.org/main/en/user/security/webadmin/passwords.html#keystore-password-providers and https://docs.geoserver.org/main/en/user/security/passwd.html#keystore-password . In my case it was the only way to get access to the admin interface as there was no admin user available (so I could not have set anything in the GUI) and I imagine that others very likely might end up in such a situation as well.

Right now a person in this situation would have zero luck finding the config.xml option with a search engine.